### PR TITLE
[Bug Fix] Fix for charges not being sold correctly

### PR DESCRIPTION
**Pull request by dencelle**
https://github.com/EQEmu/Server/issues/1350 this fixes this issue

### CODE REVIEW

**Comment by dencelle** [comment]
k, this fixes all the issues with selling / buying charged items to merchants... if a item has 0 charges i've implemented a way to properly buy items with 0 charges.

**Review by thalix1337 — APPROVED** [review]
Tested all combinations, works as intended.

**Comment by Akkadius** [comment]
> Tested all combinations, works as intended.

Thanks for being so great testing changes!

File: zone/zone.cpp [hunk 1]
```diff
@@ -419,12 +419,11 @@ int Zone::SaveTempItem(uint32 merchantid, uint32 npcid, uint32 item, int32 charg
 				if (!ml.origslot) {
 					ml.origslot = ml.slot;
 				}
-
-				if (charges > 0) {
+				bool IsStackable = database.GetItem(item)->Stackable;
```
Comment (Akkadius) [inline zone/zone.cpp:422]: Make sure to `snake_case` locally scoped variable names in the future

**Review by Akkadius — APPROVED** [review]

**Pull request merged**